Modules/Typescript/Design-Patterns/shipper.ts

```diff
@@ -0,0 +1,32 @@
+import { AirEastShipper } from './airEastShipper';
+import { PacificParcel } from './pacificParcel';
+import { ChicagoSprint } from './chicagoSprint';
+
+export class Shipper {
+  
+  weight: number;
+  zip: string;
+
+  constructor(weight: number, zip?: string) {
+    this.weight = weight;
+    this.zip = zip || '';
+  }
+
+  getCosts(): number {
+    let shipper = new AirEastShipper(this.weight);
+
+    if (this.zip) {
+      let tempShipper = new ChicagoSprint(this.weight);
+      if (tempShipper.isCarrier(this.zip)) {
+        shipper = tempShipper;
+      } else {
+        tempShipper = new PacificParcel(this.weight);
+        if (tempShipper.isCarrier(this.zip)) {
+          shipper = tempShipper;
+        }
+      }
+    }
```

Comment:
I think that this solution is really impractical, you should decide first which class to use by zipcode, not create a new instance for each check, Imagine we had 10 carriers, 50 carriers? Creating 50 new instances that each have their own checks whether they are the correct carrier doesn't sound like a good idea to me